File: frontend/types/agent.ts
export interface AgentInfo {
  name: string;
  type: string;
  description: string;
  capabilities: string[];
}

export interface AgentLog {
  id: string;
  workflow_id: string;
  step_id?: string;
  agent_type: string;
  level: 'INFO' | 'WARNING' | 'ERROR' | 'DEBUG';
  message: string;
  details: Record<string, any>;
  timestamp: string;
}

File: backend/app/schemas/agent.py
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class AgentInfo(BaseModel):
    name: str
    type: str
    description: str
    capabilities: list[str]

class AgentLogResponse(BaseModel):
    id: str
    workflow_id: str
    step_id: Optional[str]
    agent_type: str
    level: str
    message: str
    details: Dict[str, Any]
    timestamp: datetime

    model_config = ConfigDict(from_attributes=True)

class AgentLogFilter(BaseModel):
    agent_type: Optional[str] = None
    level: Optional[str] = None
    search: Optional[str] = None
    limit: int = 100
